File: 0011-container-with-most-water/0011-container-with-most-water.cpp
class Solution {
public:
    int maxArea(vector<int>& height) {
        int maxWater = 0;
        int lp = 0;
        int rp = height.size() - 1;

        while(lp < rp){
            int width = rp - lp;
            int minHeight = min(height[lp], height[rp]);
            int capacity = width * minHeight;

            maxWater = max(capacity, maxWater);

            height[lp] < height[rp] ? lp++ : rp-- ;
        }
        return maxWater;


    }
};

File: 0011-container-with-most-water/0011-container-with-most-water.java
class Solution {
    public int maxArea(int[] height) {
        int low = 0;
        int high = height.length - 1;
        int max = 0;

      while(low<high){
        int width = high - low;
        int minHeight = Math.min(height[low], height[high]);
        int capacity = width * minHeight;

        max= Math.max(max,capacity);
        if (height[low] < height[high]) {
                low++;
            } else {
                high--;
            }
      }
      return max;
    }
}